### PR TITLE
[ENG-783] As a user, I should not be allowed to add the same campaign twice for a Source.

### PR DESCRIPTION
**Pull request by dean-martin**
**Please be sure you are submitting this against the _master_ branch.**

[//]: # This Pull Request (Place an 'X' for each):

| Risk Level                                | No | Low | High |
| ----------------------------------------- | -- | --- | ---- |
| Alters Lead Data?                         |    |     |      |
| Schema Change?                            |    |     |      |
| Adds A Query or Modifies Existing Query?  |    |     |      |
| Adds or Modifies Existing Auto-Enhancer?  |    |     |      |
| Modifies Ingestion Process?               |    |     |      |
| Modifies sendContact Data?                |    |     |      |

[//]: # ( Required: )

### CODE REVIEW

File: Assets/js/03.campaigns.js [hunk 1]
```diff
@@ -68,6 +68,15 @@ Mautic.contactsourceCampaigns = function () {
...
                             if (typeof obj === 'object') {
+                                var campaignIds = [];
+                                obj.campaigns.forEach(function(cmpn) { 
+                                    if(campaignIds.indexOf(cmpn.campaignId) >= 0) {
+                                        alert("You shouldn't add the same campaign to a Source more than once, create a different campaign instead.");
+                                    }
+                                    campaignIds.push(cmpn.campaignId);
+                                });
+
+
```
Comment (heathdutton) [inline Assets/js/03.campaigns.js]: I'm afraid this isn't enough. Our users may just skip this alert... Any way we can block the addition instead?

Comment (dean-martin) [reply Assets/js/03.campaigns.js]: It now reverts the selected campaign to zero.

File: Assets/js/03.campaigns.js [hunk 2]
```diff
@@ -67,7 +67,19 @@ Mautic.contactsourceCampaigns = function () {
...
-                            if (typeof obj === 'object') {
+                            if (typeof obj === 'object') { 
+                                var campaignIds = [];
+                                for(var i = 0; i < obj.campaigns.length; i++) { 
+                                    var camp = obj.campaigns[i];
+                                    if(campaignIds.indexOf(camp.campaignId) >= 0) {
+                                        obj.campaigns[i].campaignId = "0"; 
+                                        mQuery('[name="root[campaigns][' + i + '][campaignId]"]').val(0).change();
+                                        campaignsJSONEditor.setValue(obj);
+                                        alert("You shouldn't add the same campaign to a Source more than once, create a different campaign instead.");
```
[the remark this answers is not in the document]
Comment (dean-martin) [reply Assets/js/03.campaigns.js]: I'm not sure, I wasn't able to find a Mautic JavaScript notification library in place.

[the remark this answers is not in the document]
Comment (heathdutton) [reply Assets/js/03.campaigns.js]: Scrutinizer really doesn't like this kind of thing, but it's a rare use case, so I'm ok with it for now.

**Pull request merged**